evap/student/views.py

```diff
@@ -255,31 +303,38 @@ def render_vote_page(request, evaluation, preview, for_rendering_in_modal=False)
 
 
 @participant_required
-def vote(request, evaluation_id):  # noqa: PLR0912
+def vote(request: HttpRequest, evaluation_id: int, dropout=False) -> HttpResponse:  # noqa: PLR0912
     # pylint: disable=too-many-nested-blocks
     evaluation = get_object_or_404(Evaluation, id=evaluation_id)
+
+    if dropout and not evaluation.is_dropout_allowed:
+        raise SuspiciousOperation("Drop out not allowed")
+
     if not evaluation.can_be_voted_for_by(request.user):
         raise PermissionDenied
 
-    form_groups = get_vote_page_form_groups(request, evaluation, preview=False)
+    form_groups = get_vote_page_form_groups(request, evaluation, preview=False, preselect_no_answer=False)
     if not all(form.is_valid() for form_group in form_groups.values() for form in form_group):
-        return render_vote_page(request, evaluation, preview=False)
+        return render_vote_page(request, evaluation, preview=False, dropout=dropout)
 
     # all forms are valid, begin vote operation
     with transaction.atomic():
         # add user to evaluation.voters
         # not using evaluation.voters.add(request.user) since that fails silently when done twice.
         evaluation.voters.through.objects.create(userprofile_id=request.user.pk, evaluation_id=evaluation.pk)
 
+        if dropout:
+            evaluation.dropout_count += 1
```

Comment:
As discussed in person: The evaluation is fetched before the `transaction.atomic` starts, so if two people drop out at roughly the same time, we loose one increase here